### DISCUSSION TITLE
AlaSQL grammar online editor/tester

### DISCUSSION THREAD

**Discussion by mingodad** [agershun/alasql#1806] [Show and tell]
I've just added `AlaSQL` grammar to https://mingodad.github.io/parsertl-playground/playground/ an online `Yacc/Lex` editor/tester (select `AlaSQL parser` from `Examples` then click `Parse` to see a parser tree for the contents in `Input source` editor).

The repository is here https://github.com/mingodad/parsertl-playground .

I hope it can help develop/debug/document this grammar project !

Any feedback is welcome !

**Comment by mingodad** [comment]
Doing so I noticed that `AlaSQL` doesn't accept the `insert into tbl(fld1, fld2) values(1,2);` only `insert into tbl values(1,2);`, the example shown there is the same one from `SQLite3 parser` that does accept the first form.

Also in several places I commented the grammar with `//` at the beginning of the line for the parts that are not used (apparently `jison` do not detect unused rules/tokens).

**Reply by mathiasrw** [reply]
I love that you are taking the grammar and helping detect issues. I have noted that `insert into tbl(fld1, fld2) values(1,2)` is not working. Ill make an issue with it. 

Please please let me know if you find other things. The grammar is the weakest, yet most important area in the whole alasql engine.
